cpp/src/shim/cufile.cpp

```diff
@@ -79,21 +69,6 @@ cuFileAPI::cuFileAPI()
     get_symbol(StreamRegister, lib, KVIKIO_STRINGIFY(cuFileStreamRegister));
     get_symbol(StreamDeregister, lib, KVIKIO_STRINGIFY(cuFileStreamDeregister));
   }
-
-  // cuFile is supposed to open and close the driver automatically but
-  // because of a bug in cuFile v1.4 (CUDA v11.8) it sometimes segfaults:
-  // <https://github.com/rapidsai/kvikio/issues/159>.
-  if (version < 1050) { driver_open(); }
-}
-
-// Notice, we have to close the driver at program exit (if we opened it) even though we are
-// not allowed to call CUDA after main[1]. This is because, cuFile will segfault if the
-// driver isn't closed on program exit i.e. we are doomed if we do, doomed if we don't, but
-// this seems to be the lesser of two evils.
-// [1] <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#initialization>
-cuFileAPI::~cuFileAPI()
```

Comment:
Seeing the following error on all CI jobs (like [this job]( https://github.com/rapidsai/kvikio/actions/runs/16328740410/job/46125867694?pr=770#step:10:308 )):

```cpp
./BASIC_IO_EXAMPLE: symbol lookup error: /opt/conda/envs/test/bin/tests/libkvikio/../../../lib/libkvikio.so: undefined symbol: _ZN6kvikio9cuFileAPID1Ev
```

Wondering if this is related to the destructor. Do we still want an empty constructor or do want to drop this code as well?

https://github.com/rapidsai/kvikio/blob/54c420a652bf29e3c7de9ac2e2e19af07de7c256/cpp/include/kvikio/shim/cufile.hpp#L66-L73

Reply:
Have pushed this change in commit: https://github.com/rapidsai/kvikio/pull/770/commits/53036833e74338fe9f888dc5ed6223fd2919e1bd

Reply:
Looks like that fixed CI 🎉